### DISCUSSION TITLE
Confusion about refreshing tokens and response stating 3600 for duration and refresh not stating any expiration date

### DISCUSSION THREAD

**Discussion by PdubsCryptoWorld** [etsy/open-api#1104] [Q&A]
So i've been working on the token authorization which i found a couple of functional ways to get, and i get the standard token, duration, refresh token, bearer information, now when i use the refresh token request to refresh_token i get another response that is the same thing and has the duration as 3600 again. I'm trying to understand this exactly.

I know that the api resource states that the refresh token is a 90 day. Am i understanding that when we get the refresh token and it does not have an expires at in the response to take it that our refresh_token is now good to use for 90 days even though there is no clear verification that is the case?

**Comment by wingsofred** [comment]
The refresh token is good for 90 days. When you do a "refresh token", it does give you a new refresh token but you don't have to use it. You can continue to use the one you were using for the 90 day period. The access token is the only one with the 1 hr expiration time. 

In my process, I use the same refresh token for 30 days and then use a new one. That way I don't risk ever getting close to the 90 day mark.
